### PR TITLE
Pass views by const& in fftshift

### LINKED ISSUE #191: Pass views by `const&` in `KokkosFFT::fftshift`
Following #158 we should also pass views in `KokkosFFT::fftshift` by `const&`

### PR DESCRIPTION
**Pull request by tpadioleau**
Fixes #191

### CODE REVIEW

File: common/src/KokkosFFT_Helpers.hpp
Comment (tpadioleau) [inline common/src/KokkosFFT_Helpers.hpp:1]: We are using `ViewType` to allocate arrays (`tmp`), if one passes an unmanaged view it will not compile.

Comment (yasahi-hpc) [reply common/src/KokkosFFT_Helpers.hpp:1]: That is right. Need to rely on a manageable View type.

Comment (tpadioleau) [reply common/src/KokkosFFT_Helpers.hpp:1]: Yes exactly

File: common/src/KokkosFFT_Helpers.hpp [hunk 1]
```diff
@@ -39,8 +39,8 @@ auto get_shift(const ViewType& inout, axis_type<DIM> axes, int direction = 1) {
 }
 
 template <typename ExecutionSpace, typename ViewType>
-void roll(const ExecutionSpace& exec_space, ViewType& inout, axis_type<1> shift,
-          axis_type<1>) {
+void roll(const ExecutionSpace& exec_space, const ViewType& inout,
+          axis_type<1> shift, axis_type<1>) {
```
Comment (yasahi-hpc) [inline common/src/KokkosFFT_Helpers.hpp:43]:
```suggestion
          axis_type<1> shift, axis_type<1> /* axes */) {
```

**Review by yasahi-hpc — APPROVED** [review]
Thanks, LGTM
Recently, we express the unused variable as a comment.

**Comment by tpadioleau** [comment]
> Thanks, LGTM Recently, we express the unused variable as a comment.

There is clang-tidy check to detect unnamed parameters if you want to enforce this rule.

**Comment by yasahi-hpc** [comment]
> > Thanks, LGTM Recently, we express the unused variable as a comment.
> 
> There is clang-tidy check to detect unnamed parameters if you want to enforce this rule.

This one?
https://clang.llvm.org/extra/clang-tidy/checks/readability/named-parameter.html

I will certainly add this rule

**Comment by tpadioleau** [comment]
> This one? https://clang.llvm.org/extra/clang-tidy/checks/readability/named-parameter.html
> 
> I will certainly add this rule

Yes that is it

**Comment by yasahi-hpc** [comment]
I added fixes #191 in order to close the issue as soon as this PR is merged

**Pull request merged**